contracts/smart-contract-wallet/base/ModuleManager.sol

```diff
@@ -18,11 +19,11 @@ contract ModuleManager is SelfAuthorized, Executor {
     mapping(address => address) internal modules;
 
     function setupModules(address to, bytes memory data) internal {
-        require(modules[SENTINEL_MODULES] == address(0), "BSA100");
+        if(modules[SENTINEL_MODULES] != address(0)) revert ModulesAlreadyInitialized();
         modules[SENTINEL_MODULES] = SENTINEL_MODULES;
         if (to != address(0))
             // Setup has to complete successfully or transaction fails.
-            require(execute(to, 0, data, Enum.Operation.DelegateCall, gasleft()), "BSA000");
+            if(!execute(to, 0, data, Enum.Operation.DelegateCall, gasleft())) revert ModulesSetupExecutionFailed();
```

Comment:
taking a note so it helps with resolving conflicts. we should probably add test case to catch this